core/unit_test/TestGraph.hpp

```diff
@@ -103,28 +103,30 @@ struct TEST_CATEGORY_FIXTURE(graph) : public ::testing::Test {
   }
 };
 
+/// If the counts are correct, all nodes have been submitted.
+/// However, it does not imply that dependencies were observed.
+#define ASSERT_THAT_ALL_NODES_WERE_SUBMITTED(__exec__, __expt_count__, \
+                                             __expt_bugs__)            \
```

Comment:
Even though you could argue macro arguments are not C++ identifiers, please avoid double underscores.
I typically prefer all uppercase names but we don't really have a hard convention.
"expt" is generally not an abbreviation for expected (I suppose that what you were going for) but for experiment so you might want to come up with something else.

Reply:
About the name of the macro.  It is not just that "nodes have been submitted" is it?

Reply:
- For macro arguments, my convention is...to break any convention. By adding the leading and trailing double underscore, I ensure that it's very unlikely that the macro's arguments match the name of any other variable.
- For the name, agreed. If the checks are fine, it means that the nodes also successfully executed their kernel.

Reply:
- But reading https://timsong-cpp.github.io/cppwp/lex.name#3... yeah, it might not be a good convention indeed.

Reply:
I just changed the name (and the description), and removed the `__`.